### PR TITLE
cpu: add STM32L010C6 support

### PR DESCRIPTION
**Pull request by wysman**
### Contribution description

Add support for the CPU STM32L010C6

### Testing procedure

We have test on a custom board with the following applications and our custom firmware
- tests/shell
- tests/periph_timer
- tests/periph_rtc
- Our application (using shell, timer, eeprom)

### CODE REVIEW

File: cpu/stm32_common/stm32_mem_lengths.mk [hunk 1]
```diff
@@ -190,7 +190,13 @@ ifeq ($(STM32_TYPE), F)
...
 else ifeq ($(STM32_TYPE), L)
   ifeq ($(STM32_FAMILY), 0)
-    ifneq (, $(filter $(STM32_MODEL2), 1 2))
+		ifeq ($(STM32_MODEL2), 1)
+			ifeq ($(STM32_MODEL3), 0)
+				ifeq ($(STM32_ROMSIZE), 6)
+					RAM_LEN = 8K
+				endif
+			endif
+		else ifneq (, $(filter $(STM32_MODEL2), 1 2))
```
Comment (benpicco) [inline cpu/stm32_common/stm32_mem_lengths.mk]: Please use two spaces for indentation to be consistent with the rest of the file.

File: cpu/stm32l0/include/vendor/stm32l0xx.h [hunk 2]
```diff
@@ -132,7 +132,15 @@
 /** @addtogroup Device_Included
   * @{
   */
-#if defined(STM32L011xx)
+#if defined(STM32L010xB)
+  #include "stm32l010xb.h"
+#elif defined(STM32L010x8)
+  #include "stm32l010x8.h"
+#elif defined(STM32L010x6)
```
Comment (benpicco) [inline cpu/stm32l0/include/vendor/stm32l0xx.h]: Those files are missing.

File: cpu/stm32l0/include/periph_cpu.h [hunk 3]
```diff
@@ -81,6 +82,12 @@ typedef struct {
 #define EEPROM_SIZE                (2048U)  /* 2kB */
 #elif defined(CPU_LINE_STM32L031xx)
 #define EEPROM_SIZE                (1024U)  /* 1kB */
+#elif defined(CPU_LINE_STM32L010xB) || defined(CPU_LINE_STM32L011x3) || defined(CPU_LINE_STM32L011x4) || defined(CPU_LINE_STM32L011x4)
```
Comment (benpicco) [inline cpu/stm32l0/include/periph_cpu.h]: `CPU_LINE_STM32L011x4` is listed twice, the `STM32L011x3` variant isn't handled anywhere else I think.

**Comment by benpicco** [comment]
Sorry, I didn't notice you pushed an update. 

Since you provided definitions for them, can you also add the other vendor files?
You do `#include` them, but they are not part of this PR.

File: cpu/stm32l0/include/vendor/stm32l0xx.h (hunk 2 above)
Comment (wysman) [reply cpu/stm32l0/include/vendor/stm32l0xx.h]: I have just prepare those include to help people to fine the right one in the HAL.
I don't have carrier board for thoses CPU, so i can't test this part.

Do you prefers i remove this part ?

File: cpu/stm32l0/include/periph_cpu.h (hunk 3 above)
Comment (wysman) [reply cpu/stm32l0/include/periph_cpu.h]: Yes, it's a typo for `CPU_LINE_STM32L021x4`

File: cpu/stm32l0/include/vendor/stm32l0xx.h (hunk 2 above)
Comment (benpicco) [reply cpu/stm32l0/include/vendor/stm32l0xx.h]: Adapting the vendor files to what RIOT expects is only a small patch

```patch
--- stm32l010xx.h       2019-12-13 09:46:42.085251331 +0100
+++ stm32l010xx.h       2019-12-13 09:45:43.034789432 +0100
@@ -72,7 +72,7 @@
 /******  Cortex-M0 Processor Exceptions Numbers ******************************************************/
   NonMaskableInt_IRQn         = -14,    /*!< 2 Non Maskable Interrupt                                */
   HardFault_IRQn              = -13,    /*!< 3 Cortex-M0+ Hard Fault Interrupt                       */
-  SVC_IRQn                    = -5,     /*!< 11 Cortex-M0+ SV Call Interrupt                         */
+  SVCall_IRQn                 = -5,     /*!< 11 Cortex-M0+ SV Call Interrupt                         */
   PendSV_IRQn                 = -2,     /*!< 14 Cortex-M0+ Pend SV Interrupt                         */
   SysTick_IRQn                = -1,     /*!< 15 Cortex-M0+ System Tick Interrupt                     */
 
@@ -102,7 +102,6 @@
   */
 
 #include "core_cm0plus.h"
-#include "system_stm32l0xx.h"
 #include <stdint.h>
 
 /** @addtogroup Peripheral_registers_structures
@@ -414,10 +413,7 @@
   __IO uint32_t PSC;       /*!< TIM prescaler register,                       Address offset: 0x28 */
   __IO uint32_t ARR;       /*!< TIM auto-reload register,                     Address offset: 0x2C */
   uint32_t      RESERVED12;/*!< Reserved                                      Address offset: 0x30 */
-  __IO uint32_t CCR1;      /*!< TIM capture/compare register 1,               Address offset: 0x34 */
-  __IO uint32_t CCR2;      /*!< TIM capture/compare register 2,               Address offset: 0x38 */
-  __IO uint32_t CCR3;      /*!< TIM capture/compare register 3,               Address offset: 0x3C */
-  __IO uint32_t CCR4;      /*!< TIM capture/compare register 4,               Address offset: 0x40 */
+  __IO uint32_t CCR[4];    /*!< TIM capture/compare register 1,               Address offset: 0x34 */
   uint32_t      RESERVED17;/*!< Reserved,                                     Address offset: 0x44 */
   __IO uint32_t DCR;       /*!< TIM DMA control register,                     Address offset: 0x48 */
   __IO uint32_t DMAR;      /*!< TIM DMA address for full transfer register,   Address offset: 0x4C */
```

I've crerated a PR to this branch https://github.com/Power-Lan/RIOT/pull/1

**Comment by kaspar030** [comment]
@benpicco this was not merged the usual way (squash, create merge commit). It only shows up as single commit. Did you "squash and merge"?

**Comment by benpicco** [comment]
Yes I wanted to see if this new Github feature is useful.

**Comment by aabadie** [comment]
> I wanted to see if this new Github feature is useful.

Better ask on the maintainers mailing list first. I did ask for using "squash and merge" (I let people browse the history) but the idea was rejected for obscure reasons.

**Comment by wysman** [comment]
Thanks everyone for the review

**Pull request merged**